api/s2n.h

```diff
@@ -596,6 +596,258 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
 S2N_API 
 extern int s2n_cert_get_utf8_string_from_extension_data(const uint8_t *extension_data, uint32_t extension_len, uint8_t *out_data, uint32_t *out_len);
 
+/* Pre-shared key (PSK) Hash Algorithm - RFC 8446 Section-2.2 */
+/* https://tools.ietf.org/html/rfc8446#section-2.2 */
+typedef enum {
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk;
+
+/**
+ * Creates a new s2n external pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default 
+ * pre-shared key hash algorithm. An external PSK is an out-of-band provisioned pre-shared key (PSK) established 
+ * externally using a secure mutually agreed upon mechanism. 
+ * 
+ * # Safety 
+ *
+ * Use `s2n_psk_free` to free the memory allocated to the s2n external pre-shared key object created by this API. 
+ *
+ * @return struct s2n_psk* Returns a pointer to the newly created external pre-shared key object.
+ */
+S2N_API
+struct s2n_psk* s2n_external_psk_new();
+
+/**
+ * Frees the memory associated with the `s2n_psk` object.
+ *
+ * @param psk Pointer to the pre-shared key object to be freed.
+ */
+S2N_API 
+int s2n_psk_free(struct s2n_psk **psk);
+
+/**
+ * Sets the pre-shared key (PSK) identity for a given s2n_psk object.
+ * The pre-shared key identity is a unique identifier for the pre-shared secret.
+ * It is a non-secret value represented by a UTF-8 string.
+ *
+ * # Safety 
+ *
+ * The Pre-shared key identity is transmitted over the network unencrypted and is a non-secret value.
+ * Be sure to not include any confidential information as a part of the pre-shared key identity. 
+ * 
+ * Note that it is possible for multiple clients to use the same pre-shared key secret to connect to the server 
+ * in which case they would all share the pre-shared key identity associated with the shared pre-shared key secret.
```

Comment:
Yes,  multiple clients also use the same PSK secret but with different PSK identities. The reason I mentioned this is because of the `The pre-shared key identity is a unique identifier` mentioned in the previous line, if it is confusing - I can get rid of it.